typing/ctype.ml

```diff
@@ -6773,6 +6830,14 @@ let nondep_type_decl env mid is_covariant decl =
                 Private
             with Nondep_cannot_erase _ ->
               None, decl.type_private
+    and jkind =
```

Comment:
I think @lpw25 should review these lines.

Reply:
I think it would be worth writing a proper `nondep_jkind` for l-kinds here, that weakened only the parts of the kind that referred to paths being removed when `is_covariant` is true. (This is ignoring the fact that [the whole `is_covariant` thing isn't really sound](https://github.com/ocaml/ocaml/issues/10491). But it's probably better to be consistent with the rest of the `nondep` machinery here.)

Reply:
Worth it now, in this patch? Or something we can defer until later? If "now", I'll have to learn what `nondep` is all about, something I've hitherto avoided.

Reply:
I would probably do it now if it were me, because it is probably only a few minutes work to do. But if you think it will take you a while due to needing to pick up context first then maybe leave a note and come back to it later.

Reply:
I've now given myself enough context to understand what's going on here, but I still think it will be ~hours and not ~minutes, accounting for testing, bugs, review, etc. I'm punting, but I've clarified the CR.